src/Npgsql/TypeHandling/NullableHandler.cs

```diff
@@ -14,11 +15,11 @@ namespace Npgsql.TypeHandling
 
     static class NullableHandler<T>
     {
-        public static readonly Type UnderlyingType;
-        public static readonly ReadDelegate<T> Read;
-        public static readonly ReadAsyncDelegate<T> ReadAsync;
-        public static readonly ValidateAndGetLengthDelegate<T> ValidateAndGetLength;
-        public static readonly WriteAsyncDelegate<T> WriteAsync;
+        public static readonly Type? UnderlyingType;
+        [NotNull] public static readonly ReadDelegate<T>? Read;
```

Comment:
Same comment as in Counter.cs - consider keeping this non-nullable but initializing to `null!`.

Reply:
Not here because it can be null when `UnderlyingType` is null, but there is no way to specify such dependency.

Reply:
But you're doing exactly the same by adding [NotNull] - it's just an easier way of doing the same thing.

Reply:
Probably it makes sense to remove attributes and check the field before invocation instead of the `Exists` property.

Reply:
We should not spend the whole day on it :) It's OK for the contract to say that you first check `Exists`, and if it returns true, you are guaranteed to never see a null delegate; and so the delegate is non-nullable, but still initialized to null (which again is the same thing that you get by having a nullable delegate with [NotNull]).